treon/treon.py

```diff
@@ -106,6 +107,12 @@ def print_test_collection(notebooks):
     LOG.debug(message)
 
 
+def filter_results(results, args):
+    for exclude_str in args['--exclude']:
+        results = [file for file in results if not file.startswith(exclude_str)]
```

Comment:
Should we restrict exclude_str to be valid paths? It would be easy to implement with path module.

The benefit of that is we can handle home directory symbol("~" ) etc. also as part of exclude option. It would also look more professional to accept valid paths instead of strings.

I do not have a strong opinion on this one, so I leave it upto you to make a choice.